qiskit_nature/problems/second_quantization/lattice/models/heisenberg_model.py

```diff
@@ -0,0 +1,131 @@
+        """
+        super().__init__(lattice)
+        self.coupling_constants = coupling_constants
+        self.ext_magnetic_field = ext_magnetic_field
+
+    def coupling_matrix(self) -> np.ndarray:
+        """Return the coupling matrix."""
+        return self.interaction_matrix()
+
+    @classmethod
+    def uniform_parameters(
+        cls, lattice: Lattice, uniform_interaction: complex, uniform_onsite_potential: complex
+    ) -> "HeisenbergModel":
+        """Set a uniform interaction parameter and on-site potential over the input lattice.
+
+        Args:
+            lattice (Lattice): Lattice on which the model is defined.
+            uniform_interaction (complex): The interaction parameter.
+            uniform_onsite_potential (complex): The on-site potential.
+
+        Returns:
+            HeisenbergModel: The Heisenberg model with uniform parameters.
+        """
+        return cls(
+            cls._generate_lattice_from_uniform_parameters(
+                lattice, uniform_interaction, uniform_onsite_potential
+            )
+        )
+
+    @classmethod
+    def from_parameters(cls, interaction_matrix: np.ndarray) -> "HeisenbergModel":
+        """Return the Hamiltonian of the Heisenberg model
+        from the given interaction matrix and on-site interaction.
+
+        Args:
+            interaction_matrix (np.ndarray): A real or complex valued squared matrix.
+
+        Returns:
+            HeisenbergModel: The Heisenberg model generated from the given interaction
+                matrix and on-site interaction.
+        """
+        return cls(cls._generate_lattice_from_parameters(interaction_matrix))
+
+    def second_q_ops(
+        self,
+        display_format: Optional[str] = None,
+    ) -> SpinOp:
+        """Return the Hamiltonian of the Heisenberg model in terms of `SpinOp`.
+
+        Args:
+            display_format (Optional[str], optional): Not supported for Spin operators. If specified, it will be ignored. Defaults to None.
+
+        Returns:
+            SpinOp: The Hamiltonian of the Heisenberg model.
+        """
+        if display_format is not None:
+            logger.warning(
+                "Spin operators do not support display-format. Provided display-format "
+                "parameter will be ignored."
+            )
+
+        hamiltonian = []
+        weighted_edge_list = self.lattice.weighted_edge_list
+        register_length = self.lattice.num_nodes
+
+        for node_a, node_b, _ in weighted_edge_list:
+
+            if node_a == node_b:
+                index = node_a
+                if self.ext_magnetic_field[0] != 0:
+                    hamiltonian.append((f"X_{index}", self.ext_magnetic_field[0]))
+                if self.ext_magnetic_field[1] != 0:
+                    hamiltonian.append((f"Y_{index}", self.ext_magnetic_field[1]))
+                if self.ext_magnetic_field[2] != 0:
+                    hamiltonian.append((f"Z_{index}", self.ext_magnetic_field[2]))
```

Comment:
May I suggest smth like the following:
```python
for axis, coeff in zip("XYZ", self.ext_magnetic_field):
    if not np.isclose(coeff, 0.0):
        hamiltonian.append((f"{axis}_{index}", coeff))
```

You can do the same in the case of the `coupling_constants`

Reply:
Much better! Thanks for this code suggestion